docs/TuxStores.md

```diff
@@ -0,0 +1,196 @@
+Event string to add a listener callback to. If no event is passed in than the listener will be added to the default `CHANGE` event.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    var newEventType = 'NEW_EVENT_TYPE';
+    store.addChangeListener(listenerCallback, newEventType);
+```
+
+
+### <a id="store-removeChangeListener"></a>Method `store.removeChangeListener` - type: FUNCTION [#](#store-removeChangeListener)
+Removes the passed in listener callback from the passed in event string. If no event is specified than the default `CHANGE` event is used.
+
+```
+    store.removeChangeListener(callback, EVENT_TYPE);
+```
+
+##### <a id="removeChangeListener-callback"></a>Parameter - `callback` - type: FUNCTION - required [#](#removeChangeListener-callback)
+The callback function that will be removed from the store's set of listeners.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    store.addChangeListener(listenerCallback);
+    store.removeChangeListener(listenerCallback);
+```
+
+##### <a id="removeChangeListener-EVENT_TYPE"></a>Parameter - `EVENT_TYPE` - type: STRING - optional [#](#removeChangeListener-EVENT_TYPE)
+Change event type string on the store that the listener will be removed from. If no `EVENT_TYPE` is specified the default value of `CHANGE` will be used.
+
+```
+    var listenerCallback = function () {
+      this.setState({ messages: MessageStore.all() });
+    };
+
+    var newEventType = 'NEW_EVENT_TYPE';
+    store.addChangeListener(listenerCallback, newEventType);
+    store.removeChangeListener(listenerCallback, newEventType);
+```
+
+## <a id="TuxStores-Complete-Example"></a>TuxStores Complete Example [#](#TuxStores-Complete-Example)
+Taking everything we have learned lets look at a more realistic example of using TuxStores.
+
+```
+    var Stores = require('tux/Stores');
+
+    var messageStore = Stores.createStore({
+      _messages: [],
+
+      allMessages: function () {
+        return this._messages;
+      },
+
+      onGet: function (data) {
+        this._messages = data.messages;
+        this.emitChange();
+      },
+
+      onCreate: function (message) {
+        this._messages.push(message);
+        this.emitChange();
+      },
+
+      onDelete: function (message) {
+        this._messages.splice(message.id, 1);
+        this.emitChange();
+      }
+    });
+
+    console.assert(messageStore.hasOwnProperty('emitChange'));
+    console.assert(messageStore.hasOwnProperty('addChangeListener'));
+    console.assert(messageStore.hasOwnProperty('removeChangeListener'));
+
+    var messageStoreListener = {
+      onChangeEvent: function () {
+        messageStore.getAll();
+      }
+    };
+
+    messageStore.addChangeListener(messageStoreListener.onChangeEvent);
```

Comment:
Hmm looking at this now and taking Bianca's comments into account this would probably be better as a real example.  Try this:

```
    //in Tux we have better and more semantic methods for performing the below set of operations but lets pretend we need to do things the old fashioned way for posterity's sake

    var React = require('tux/React');
    var Messages = require('./Messages');

    var MessageView = React.createOwnerClass({
      getInitialState: function () {
        return {
          messages: messageStore.getAll()
        };
      },

      listenerCallback: function () {
        this.setState({
          messages: messageStore.getAll()
        });
      },

      componentDidMount: function () {
        this.listenerCallback = this.listenerCallback.bind(this); //need to do this so we make sure listenerCallback is invoked with the proper context while it is in scope
        messageStore.addChangeListener(this.listenerCallback);
      },

      componentWillUnmount: function () {
        messageStore.removeChangeListener(this.listenerCallback);
      },

      render: function () {
        return (
          <Messages messages={this.state.messages} />
        );
      }
    };
```